### PR TITLE
Dbatiste/backdrop improvements

### PR DESCRIPTION
**Pull request by dbatiste**
Improvements:

* updated to use property so flow would be same whether setting using attribute or property
* use isVisible helper instead of offsetParent since offsetParent is null whether position is fixed
* updated to honor reduce motion preference
* add the regular stuff - readme, demo
* add visual-diff test that uses reduce motion media query

### CODE REVIEW

File: components/backdrop/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+<div id="target"><button>toggle backdrop</button></div>
+<d2l-backdrop for-target="target" shown></d2l-backdrop>
+```
+
+Set the visible state of the backdrop by using the `shown` attribute/property.
+
+```javascript
+button.addEventListener('click', () => {
+  backdrop.shown = true;
```
Comment (dlockhart) [inline components/backdrop/README.md:23]: If it's toggling, shouldn't this be `backdrop.shown = !backdrop.shown;`?

File: components/backdrop/backdrop.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+
 const scrollKeys = [];
 let scrollOverflow = null;
 
 class Backdrop extends LitElement {
 
 	static get properties() {
 		return {
-			shown: { type: Boolean },
+			shown: { type: Boolean, attribute: true },
```
Comment (dlockhart) [inline components/backdrop/backdrop.js:16]: Is this needed? I thought observers were created automatically and you only needed to set this if you wanted to rename the attribute or not have an observer.

File: components/backdrop/test/backdrop.visual-diff.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+describe('d2l-backdrop', () => {
+
+	const visualDiff = new VisualDiff('backdrop', __dirname);
+
+	let browser, page;
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+		await page.emulateMediaFeatures([{ name: 'prefers-reduced-motion', value: 'reduce' }]);
```
Comment (dlockhart) [inline components/backdrop/test/backdrop.visual-diff.js:13]: Probably don't want everyone to have to set this in all their tests (they'll forget, or do it inconsistently)... is there an easier way we could expose this for everyone?

File: components/backdrop/backdrop.js (hunk 2 above)
Comment (dbatiste) [reply components/backdrop/backdrop.js:16]: I just tried this out again and it appear to work without... will remove.  I must have added while having some trouble and thought that it was needed.

File: components/backdrop/test/backdrop.visual-diff.js (hunk 3 above)
Comment (dbatiste) [reply components/backdrop/test/backdrop.visual-diff.js:13]: 🤔 of some options but they all have issues.  Perhaps lets discuss tomorrow if you have time.  :)

**Pull request merged**